### PR TITLE
New post: Building Rust Applications For AWS Graviton

### PR DESCRIPTION
**Pull request by DDxPlague**
# Content Pull Request Checklist

Please check to make sure your content submission meets the following requirements by adding an `x` between the brackets:

- [ X] The title of this pull request reflects the title of my content.
- [ X] I have reviewed the [Content Review Checklist](https://github.com/build-on-aws/content/blob/main/CONTENT_REVIEW_CHECKLIST.md).
- [X ] This pull request contains one piece of content (post or page).

\

**By submitting this pull request, I confirm I own this content, have permission to use each image included in this content, and that it will be released under the [CC BY-SA 4.0 SA License](/LICENSE).**

### CODE REVIEW

File: tutorials/building-rust-applications-for-aws-graviton/index.md [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+
+- How to build a Rust application for AWS Graviton
+- How to port an existing Rust application to AWS Graviton
+
+| Attributes                |                                   |
+| ------------------- | -------------------------------------- |
+| ✅ AWS Level        | 200 - Intermediate                          |
+| ⏱ Time to complete  | 30 minutes                             |
+| 💰 Cost to complete | Free when using the AWS Free Tier or USD 2.62      |
+| 🧩 Prerequisites    | - [AWS Account](https://aws.amazon.com/resources/create-account/?sc_channel=el&sc_campaign=tutorial&sc_content=building-rust-applications-for-aws-graviton&sc_geo=mult&sc_country=mult&sc_outcome=acq)<br> - [AWS DynamoDB Table](https://us-west-2.console.aws.amazon.com/dynamodbv2/home?region=us-west-2#dashboard)|
```
Comment (gaonkarr) [inline tutorials/building-rust-applications-for-aws-graviton/index.md:32]: Editorial reviewer pls update `sc_campaign=costwave` in the tracking link

File: tutorials/building-rust-applications-for-aws-graviton/index.md [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+| ToC |
+|-----|
+
+## Setup
+
+### EC2 Setup
+
+To demonstrate how to move a Rust application to AWS Graviton Instances I have built a simple link shortener in Rust. I’m not a front end developer, so I’ll be relying on cURL to interact with my application’s APIs. The application is written with the most current release version of Rust at the time of writing, and Rocket 0.5.0-rc3. The application generates a unique 8 character string for each URL that it shortens, and stores the original URL and the 8 character string in a DynamoDB table. The code is not meant to be used in production and is provided as a sample only.
+
+For this demo I’ll be using two EC2 instances running Ubuntu 22.04. The first instance will be a `c5.xlarge` instance, and the second will be a `c6g.xlarge` instance. To get started, log in to each instance and download a copy of the code. To install Rust, use the following command:
```
Comment (gaonkarr) [inline tutorials/building-rust-applications-for-aws-graviton/index.md:46]: I believe the code is download in the next step?

File: tutorials/building-rust-applications-for-aws-graviton/index.md [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+
+You should get output that looks like the following:
+
+```shell
+Your full URL is https://aws.amazon.com/ec2/graviton/
+```
+
+## Performance Testing Results
+
+Understanding the request per second each instance could process under heavy load is often a key metric used to evaluate instance and application performance. In order to compare a `c6g.xlarge` and a `c5.xlarge` instances we need to perform a load test. We discuss various load testing methodologies in the [Graviton Technical Guide on GitHub](https://github.com/aws/aws-graviton-getting-started) and recommend using a framework like [wrk2](https://github.com/kinvolk/wrk2). I decided to go ahead and use `wrk2` to test the `shorten_url` function of our application.
```
Comment (gaonkarr) [inline tutorials/building-rust-applications-for-aws-graviton/index.md:144]: The first sentence, "Understanding the request per second each instance could process under heavy load is often a key metric used to evaluate instance and application performance." is very confusing. Can we please simplify it and/or break it down ?

Comment (gaonkarr) [inline tutorials/building-rust-applications-for-aws-graviton/index.md:144]: Also, in order to compare what? performance? speed? sentence seems to be missing what we are comparing.  "In order to compare a `c6g.xlarge` and a `c5.xlarge` instances we need to perform a load test."

Comment (gaonkarr) [inline tutorials/building-rust-applications-for-aws-graviton/index.md:144]: Please add some information about wrk2 in brief,

File: tutorials/building-rust-applications-for-aws-graviton/index.md [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+|99.99 |26.02ms |26.02ms |
+|99.999 |26.02ms |26.02ms |
+|100 |26.04ms |26.02ms |
+
+| |C5.xlarge |C6g.xlarge |
+|--- |--- |--- |
+|Requests/Second |11,947.12 |11,960.24 |
+|Total Requests Served |21,504,877 |21,528,469 |
+
+Using the same load test our AWS Graviton powered instance achieved similar request latencies across all percentiles, while serving a large number of requests per second and a larger number of total requests. Our AWS Graviton instance did all of this while being cheaper and consuming less energy. AWS Graviton instances achieve the best price/performance for Rust applications.
```
Comment (gaonkarr) [inline tutorials/building-rust-applications-for-aws-graviton/index.md:182]: can we provide a little more information, atleast % of how much price performance and energy use.

File: tutorials/building-rust-applications-for-aws-graviton/index.md [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+  - ec2
+  - tutorials
+  - aws
+showInHomeFeed: true
+authorGithubAlias: DDxPlague
+authorName: Tyler Jones
+date: 2023-06-15
+---
+
+Companies today are making sustainability a key goal for their business in order to improve operational efficiency and drive down cost while also lowering carbon emissions. Achieving these sustainability goals means change across all levels of the business, with application and software development being a key focus. With Rust applications, one of the easiest ways to make progress towards a sustainability goal is to adopt AWS Graviton instances. In this tutorial I will walk through the steps to take an existing application running on x86 instances today and migrate to AWS Graviton powered instances in order to achieve a higher level of sustainability for your Rust application. This guide includes creating AWS resources that you will be charged for.
```
Comment (gaonkarr) [inline tutorials/building-rust-applications-for-aws-graviton/index.md:20]: As this is L200, please add more to the introduction. 

Please add a bit more detail about AWS Graviton and its benefits. Also link it to website or documentation.

File: tutorials/building-rust-applications-for-aws-graviton/index.md [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+
+## What you will learn
+
+- How to build a Rust application for AWS Graviton
+- How to port an existing Rust application to AWS Graviton
```
Comment (gaonkarr) [inline tutorials/building-rust-applications-for-aws-graviton/index.md:25]: I am unable to understand where and how are we guiding to migrate existing application in this tutorial? Can we be more explicit?

File: tutorials/building-rust-applications-for-aws-graviton/index.md [hunk 7]
```diff
@@ -0,0 +1,190 @@
+---
+title: Building Rust Applications For AWS Graviton
+description: Learn how to migrate a Rust application from x86 based Amazon EC2 Instances to ARM64-based Graviton EC2 Instances
```
Comment (gaonkarr) [inline tutorials/building-rust-applications-for-aws-graviton/index.md:3]: can we add some more details about benefits here? example - learn how to do do X and Y to reduce environmental impact or Learn how to do x and y to reach your sustainability goal. Or do you want to focus on both cost and sustainability? then we should showcase that. As discussed, we want to make sure we address the customer pain and show how aws service can help mitigate those. Can we add that here and/or in the introduction below?

**Comment by gaonkarr** [comment]
@DDxPlague can you please have a look at the comments? I also made some minor changes, so please pull the changes before making any new changes. also, confirm the changes I made are alright with you.

File: tutorials/building-rust-applications-for-aws-graviton/index.md (hunk 6 above)
Comment (DDxPlague) [reply tutorials/building-rust-applications-for-aws-graviton/index.md:27]: For this piece we're taking an existing application that is already written in Rust and showcasing how easy it is to move. It is deceptively simple, and something that seems more complex than it is at first glance. All you have to do is build the application on an arm64 machine and you're ready to go.

File: tutorials/building-rust-applications-for-aws-graviton/index.md (hunk 7 above)
Comment (DDxPlague) [reply tutorials/building-rust-applications-for-aws-graviton/index.md:3]: I'm happy to add some more details to the description/or introduction. I initially had some stuff in there but was told it was too "salesy" and to take it out.

File: tutorials/building-rust-applications-for-aws-graviton/index.md (hunk 2 above)
Comment (DDxPlague) [reply tutorials/building-rust-applications-for-aws-graviton/index.md]: Yes. Do I need to make any changes here?

File: tutorials/building-rust-applications-for-aws-graviton/index.md (hunk 3 above)
Comment (DDxPlague) [reply tutorials/building-rust-applications-for-aws-graviton/index.md:150]: Fixed this section in my next commit.

Comment (DDxPlague) [reply tutorials/building-rust-applications-for-aws-graviton/index.md:150]: Cleaned up this section in my next commit.

File: tutorials/building-rust-applications-for-aws-graviton/index.md (hunk 4 above)
Comment (DDxPlague) [reply tutorials/building-rust-applications-for-aws-graviton/index.md:190]: Done!

File: tutorials/building-rust-applications-for-aws-graviton/index.md (hunk 5 above)
Comment (DDxPlague) [reply tutorials/building-rust-applications-for-aws-graviton/index.md]: Done! Added a few blurbs along with statistics in the performance testing summary bits.

File: tutorials/building-rust-applications-for-aws-graviton/index.md (hunk 7 above)
Comment (DDxPlague) [reply tutorials/building-rust-applications-for-aws-graviton/index.md:3]: Added a bit in here. Let me know if that works.

**Comment by DDxPlague** [comment]
@gaonkarr I've made changes as requested. Please review and let me know what you think!

**Review by cobusbernard — APPROVED** [review]

**Pull request merged**